frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -296,9 +341,17 @@ function ReaderFooter:init()
             end
         end
     else
+        if not self.settings.max_possible_modes then
+            self.settings.max_possible_modes = 13
```

Comment:
I guess you want to force add these new items to the user already chosen set of items.
It's good for advertising the new stuff, but I'm not sure it's a really friendly decision :) I would just not mess with it, people will discover them if they need them.

Reply:
No :)
This is needed to see new new items (title and chapter).
Look at sample settings before this PR:"

```
    ["footer"] = {
        ["order"] = {
            [0] = "off",
            [1] = "page_progress",
            [2] = "time",
            [3] = "pages_left",
            [4] = "battery",
            [5] = "percentage",
            [6] = "book_time_to_read",
            [7] = "chapter_time_to_read",
            [8] = "frontlight",
            [9] = "mem_usage",
        },
[...]
```
and code:

```
    if not self.settings.order then
        self.mode_nb = 0
        self.mode_index = {}
        local mode_tbl = {}
        for k,v in pairs(MODE) do
            mode_tbl[v] = k
        end
        local mode_name
        for i = 0, #mode_tbl do
            mode_name = mode_tbl[i]
            if mode_name == "wifi_status" and not Device:isAndroid() then
                do end -- luacheck: ignore 541
            elseif mode_name == "frontlight" and not Device:hasFrontlight() then
                do end -- luacheck: ignore 541
            else
                self.mode_index[self.mode_nb] = mode_name
                self.mode_nb = self.mode_nb + 1
            end
        end
    else
        self.mode_index = self.settings.order
        self.mode_nb = #self.mode_index
    end

```
and almost always we run `else` statement and `self.mode_index` take `self.settings.order` so we see only old modes (without title and chapter). When we need to add new ones we need to add flag `self.settings.max_possilble_modes`. In the future after adding new modes we need increase max_possible_modes and 
`self.settings.order[self.mode_nb+1] = "whatever"`

I hope I explained clearly :)

Reply:
You did, thanks :)
OK, so that's just the ordering, and they won't be enabled be default?
I get that when ordering, the user orders all the items, so yes, the order table contains them all.
But it looks like you don't need `self.settings.max_possible_modes` and neither that hardcoded `13`, as you have all the existing items in `local MODE = {` on line 28. So, may be something like:
```
while #self.settings.order < #MODE do
  self.settings.order[#self.settings.order+1] = MODE[#self.settings.order+1]
end
```
but ok, with that, you're not skipping wifi_status and frontlight...
So, may be just merge these 2 branches (the rare if and the almost always else), just using the if code, and you would skip the ones already in:
```diff
        for i = 0, #mode_tbl do
            mode_name = mode_tbl[i]
            if mode_name == "wifi_status" and not Device:isAndroid() then
                do end -- luacheck: ignore 541
            elseif mode_name == "frontlight" and not Device:hasFrontlight() then
                do end -- luacheck: ignore 541
+            elseif self.settings.order and i+1 < #self.settings.order then
+                -- already in
+                do end -- luacheck: ignore 541
            else
                self.mode_index[self.mode_nb] = mode_name
                self.mode_nb = self.mode_nb + 1
            end
        end
```

Or may be keep the first solution above, and just have MODE cleaned up at top of file, just after `local MODE`:
```
if not Device:isAndroid() then
    MODE.wifi_status = nil
end
if not Device:hasFrontlight() then
    MODE.frontlight = nil
end
```
Or something like that :)
(I just want to not have `settings.max_possible_modes` and neither that hardcoded `13`)

Reply:
Done in https://github.com/koreader/koreader/pull/5775/commits/2aef4ef5d60eb0a240ed60604dbc80a3f6dfdfeb